storages/storages.go

```diff
@@ -28,7 +28,11 @@ func (be *backendError) Err() error {
 }
 
 func (be *backendError) Error() string {
-	return fmt.Sprintf("backend %s responded with error %s", be.backend, be.origErr)
+	if transportDefinitionError, ok := be.origErr.(*transport.DefinitionError); ok {
+		return fmt.Sprintf("backend %s not responded with reason: %s", be.backend, transportDefinitionError.Error())
```

Comment:
Do we need this message at all? Not matched transport has nothing to do with backend itself.